pkg/controller/deployment_inplace.go

```diff
@@ -0,0 +1,584 @@
+		return scaled, err
+	}
+
+	addedNewReplicasCount := int32(0)
+
+	for _, machineSet := range oldMachineSets {
+		transferredMachineCount := int32(0)
+		// get the machines for the machine set
+		oldMachines, err := dc.machineLister.List(labels.SelectorFromSet(machineSet.Spec.Selector.MatchLabels))
+		if err != nil {
+			return false, err
+		}
+
+		klog.V(3).Infof("Found %d machine(s) in old machine set %s", len(oldMachines), machineSet.Name)
+
+		for _, oldMachine := range oldMachines {
+			nodeName, ok := oldMachine.Labels[v1alpha1.NodeLabelKey]
+			if !ok {
+				return false, fmt.Errorf("node label not found for machine %s", oldMachine.Name)
+			}
+
+			node, err := dc.nodeLister.Get(nodeName)
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					klog.Warningf("Node %s not found for machine %s", nodeName, oldMachine.Name)
+					continue
+				}
+				return false, fmt.Errorf("failed to get node %s for machine %s: %w", nodeName, oldMachine.Name, err)
+			}
+
+			cond := getMachineCondition(oldMachine, v1alpha1.NodeInPlaceUpdate)
+			if isUpdateNotSuccessful(cond, node.Labels) {
+				continue
+			}
+
+			klog.V(3).Infof("Attempting to transfer machine %s to new machine set %s", oldMachine.Name, newMachineSet.Name)
+
+			// removes labels not present in newMachineSet so that the machine is not selected by the old machine set
+			machineNewLabels := MergeStringMaps(
+				MergeWithOverwriteAndFilter(oldMachine.Labels, machineSet.Spec.Selector.MatchLabels, newMachineSet.Spec.Selector.MatchLabels),
+				map[string]string{v1alpha1.LabelKeyNodeUpdateResult: v1alpha1.LabelValueNodeUpdateSuccessful})
+
+			formattedLabels, err := labelsutil.GetFormattedLabels(machineNewLabels)
+			if err != nil {
+				return false, err
+			}
+			// update the owner reference of the machine to the new machine set and update the labels
+			addControllerPatch := fmt.Sprintf(
+				`{"metadata":{"ownerReferences":[{"apiVersion":"machine.sapcloud.io/v1alpha1","kind":"%s","name":"%s","uid":"%s","controller":true,"blockOwnerDeletion":true}],"labels":%s,"uid":"%s"}}`,
+				v1alpha1.SchemeGroupVersion.WithKind("MachineSet"),
+				newMachineSet.GetName(), newMachineSet.GetUID(), string(formattedLabels), oldMachine.UID)
+
+			err = dc.machineControl.PatchMachine(ctx, oldMachine.Namespace, oldMachine.Name, []byte(addControllerPatch))
+			if err != nil {
+				klog.Errorf("failed to transfer the ownership of machine %s to new machine set. Err: %v", oldMachine.Name, err)
+				return false, err
+			}
+
+			// uncordon the node since the ownership of the machine has been transferred to the new machine set.
+			node.Spec.Unschedulable = false
+			_, err = dc.targetCoreClient.CoreV1().Nodes().Update(ctx, node, metav1.UpdateOptions{})
+			if err != nil {
+				return false, fmt.Errorf("failed to uncordon the node %s: %w", node.Name, err)
+			}
+
+			transferredMachineCount++ // scale down the old machine set.
+			addedNewReplicasCount++   // scale up the new machine set.
```

Comment:
both `transferredMachineCount` and `addedNewReplicasCount` are incremented together. In which situation will they have different values. If they will always have identical values then why not just use one variable?

Reply:
`transferredMachineCount ` is defined per old machineset. For every old machine set `transferredMachineCount` will count the number of machines which are transferred to the new machineset and by this number the corresponding old machinset needs to be scaled down. We are assuming there can be multiple machine sets, hence the loop runs for all the old machinsets. `addedNewReplicasCount` replica count hold the value of total number of machines transferred to the new machineset (sum of machines transferred from all the old machine set) and new machineset needs to be scaled up by this amount.